packages/@react-spectrum/s2/stories/CoachMark.stories.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {
+  ActionButton, ActionMenu,
+  Button,
+  CardPreview,
+  Checkbox,
+  CoachMark,
+  CoachMarkTrigger,
+  Content, Footer, Image, Keyboard, MenuItem,
+  Slider,
+  Text, useTour
+} from '../src';
+import Filter from '../s2wf-icons/S2_Icon_Filter_20_N.svg';
+import type {Meta} from '@storybook/react';
+import {Placement} from '@react-types/overlays';
+import {style} from '../style' with {type: 'macro'};
+import TextAdd from '../s2wf-icons/S2_Icon_TextAdd_20_N.svg';
+import TextBold from '../s2wf-icons/S2_Icon_TextBold_20_N.svg';
+import TextIcon from '../s2wf-icons/S2_Icon_Text_20_N.svg';
+import {Toolbar} from 'react-aria-components';
+
+const meta: Meta<typeof CoachMark> = {
+  component: CoachMark,
+  parameters: {
+    layout: 'centered'
+  },
+  tags: ['autodocs'],
+  argTypes: {
+    placement: {
+      control: 'radio',
+      options: ['top', 'left', 'left top', 'right', 'right top', 'bottom']
+    }
+  },
+  title: 'CoachMark'
+};
+
+export default meta;
+
+export const CoachMarkExample = (args) => (
+  <CoachMarkTrigger isOpen>
```

Comment:
I don't think this is the API we want. Otherwise Coachmarks are always in the JSX and it becomes hard to disentangle them or coordinate over an entire application for a tour. For instance, a Coachmark may not be rendered until a lazy component has loaded or maybe a view is swapped out part way through the tour.

Instead, we should use the same strategy from v2 and attach to selectors with a retry pattern for finding the current step.